docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx

```diff
@@ -0,0 +1,178 @@
+---
+title: Pre Call Diagnostics
+nav: 11.4
+---
+
+## Pre Call Diagnostic APIs
+
+The Pre Call Diagnostic APIs provide two type of functionalities :
+
+1. APIs to test the connected input and output devices like camera, mic and speaker functionality
+2. APIs to test the connectivity to 100ms Servers - This can identify signaling and media connectivity issues and provide a report at the end of the test.
+
+You can use these APIs in your application to detect issues before a Participant joins a video Room or as part of a troubleshooting page while already joined in the room or at a later stage to debug some issues at the end user's device/network
+
+## How to use the Pre Call Diagnostic APIs
+
+To use the diagnostic APIs first you need to initialize an instance of `HMSDiagnostics` from the `HMSActions` instance.
+
+```js
+// get the HMSSDK instance
+const diagnostics = hmsActions.initDiagnostics();
+```
+
+## Request Permissions
+
+If you want a list of devices to choose from before testing your devices you'll need to request corresponding permissions using
+
+```js
+diagnostics?.requestPermission({ audio: true, video: true }).catch((error) => setError(error));
+```
+
+## Test microphone
+
+To test if the microphone is correctly working, call the `startMicCheck` API passing preffered check duration as `timeInMillis` parameter. Diagnostics SDK will record the microphone input into a temporary file.
+
+You can use the `selectTrackAudioByID` selector to get the input audio level during recording
+
+```js
+hmsDiagnostics?.startMicCheck({
+    inputDevice,
+    onError:,
+    onStop,
+    time, // in ms, default 10 seconds
+});
+```
+
+You can call `stopMicCheck` if you need to interrupt the check before the specified time has passed.
+
+## Test speaker
+
+You can check if the speaker is working correctly if you are able to properly play audio using the `HTMLAudioElement` by using the pre-recorded audio file from `startMicCheck` using the `getRecordedAudio` API or use the test audio from `https://100ms.live/test-audio.wav` if such pre-recorded audio is not available.
+
+```js
+const element = new HTMLAudioElement();
+element.src = hmsDiagnostics?.getRecordedAudio() || 'https://100ms.live/test-audio.wav';
+if (typeof element.setSinkId !== 'undefined') {
+    // @ts-ignore
+    const { audioOutputDeviceId } = hmsStore.getState(selectLocalMediaSettings);
+    element.setSinkId(audioOutputDeviceId);
+}
+element.play();
+```
+
+You can set the audio output device ID using HMSActions like `hmsActions.setAudioOutputDevice(deviceId);`
```

Comment:
`HMSActions`

Reply:
The JS convention is lowercased-camelcase

Reply:
Oh I meant that HMSActions wasn't enclosed in code quote in the sentence.